source/eventHandler.py

```diff
@@ -294,19 +295,49 @@ def executeEvent(
 	):
 		return
 	try:
+		global virtualDesktopName
 		isGainFocus = eventName == "gainFocus"
 		# Allow NVDAObjects to redirect focus events to another object of their choosing.
 		if isGainFocus and obj.focusRedirect:
-			obj=obj.focusRedirect
-		sleepMode=obj.sleepMode
+			obj = obj.focusRedirect
+		sleepMode = obj.sleepMode
+		# Handle possible virtual desktop name change event.
+		# More effective in Windows 10 Version 1903 and later.
+		if (
+				eventName == "nameChange"
+				and obj.windowClassName == "#32769"
+				and canAnnounceVirtualDesktopNames
+		):
+			import core
+			virtualDesktopName = obj.name
+			core.callLater(250, handlePossibleDesktopNameChange)
 		if isGainFocus and not doPreGainFocus(obj, sleepMode=sleepMode):
 			return
-		elif not sleepMode and eventName=="documentLoadComplete" and not doPreDocumentLoadComplete(obj):
+		elif not sleepMode and eventName == "documentLoadComplete" and not doPreDocumentLoadComplete(obj):
 			return
 		elif not sleepMode:
-			_EventExecuter(eventName,obj,kwargs)
-	except:
-		log.exception("error executing event: %s on %s with extra args of %s"%(eventName,obj,kwargs))
+			_EventExecuter(eventName, obj, kwargs)
+	except Exception:
+		log.exception(f"error executing event: {eventName} on {obj} with extra args of {kwargs}")
+
+
+virtualDesktopName: Optional[str] = None

```

Comment:
This fails, because type hints can't be added to global names.

Reply:
type hints work fine for global names for me with VS Code - can you explain where this is coming from?
These are used regularly in NVDA

Reply:
To clarify, the following exception is raised when I attempt to run this PR:

```python traceback
Traceback (most recent call last):
  File "nvda.pyw", line 406, in <module>
    core.main()
  File "core.py", line 510, in main
    import NVDAHelper
  File "NVDAHelper.py", line 27, in <module>
    import eventHandler
  File "eventHandler.py", line 324
    virtualDesktopName: Optional[str] = None
                                           ^
SyntaxError: annotated name 'virtualDesktopName' can't be global
```

Reply:
this is a scoping problem. If the line is moved higher in the file, above the first `global` declaration the code runs fine.

Reply:
>
Hi, I see. I’m sure we have other files with this issue, so I think it would be great to resolve these as part of 2024.1 development cycle. Thanks.

I think it's unlikely we have other files with this issue. This bug prevents NVDA from starting if affected.